aspnetcore/signalr/configuration.md

```diff
@@ -0,0 +1,202 @@
+    {
+        hubOptions.EnableDetailedErrors = true;
+        hubOptions.KeepAliveInterval = TimeSpan.FromMinutes(1);
+    })
+}
+```
+
+Strongly-typed [HubOptions<T>](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.signalr.huboptions-1) can be configured as follows, for example, setting the `EnableDetailedErrors` property:
+
+```csharp
+  services.AddSignalR().AddHubOptions<HubName>(options =>
+  {
+      options.EnableDetailedErrors = true;
+  }
+```
+
+### HttpConnectionDispatcherOptions
+
+Options related to the transport layer. Use these to restrict the transports that can be used by SignalR clients, as well as to configure advanced settings related to memory buffer management. These options are configured by passing a delegate to `MapHub<T>`.
+
+| Option | Description |
+| ------ | ----------- |
+| `ApplicationMaxBufferSize`  | The maximum number of bytes the connection (transport) can buffer when invoking methods from the client, before blocking and waiting for the application to consume enough bytes to allow writing again.  |
+| `AuthorizationData` | A pre-populated list of `IAuthorizeData` gathered from the `Authorize` attributes used on the `Hub<T>`. |
+| `LongPolling`  | Gets the LongPolling options object that has a settable `PollTimeout` property. Setting the `PollTimeOut` sets the wait time before ending a poll request. |
+| `TransportMaxBufferSize`  | The maximum number of bytes the application. For example, `Clients.All.SendAsync` can buffer when writing to a single connection before blocking and waiting for the connection to consume enough bytes to allow writing again. |
+| `Transports`  | A bitmask that sets a transport. The available `options.Transports` options are as follows: `HttpTransportType.WebSockets`, `HttpTransportType.LongPolling`, and `HttpTransportType.ServerSentEvents` |
+| `WebSockets`  |  Gets the WebSockets options object.  |
+| `CloseTimeout`  | TimeSpan to set how long to wait for a clean WebSocket close when connection is being terminated  |
+| `SubProtocolSelector` | A delegate the hub calls and passes a list of `Sec-WebSocket-Protocol` values from the request header. A delegate returns the chosen `Sec-WebSocket-Protocol`. |
+| AccessTokenProvider | A delegate that is called to get an access token. The token is applied as an HTTP Bearer Authentication header. |
+
+### [HubConnectionBuilder](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.signalr.client.hubconnectionbuilder?view=aspnetcore-2.1)
+
+The `HubConnectionBuilder` API is available for both C# and TypeScript clients.
+
+| Option | Description |
+| ------ | ----------- |
+| Headers | HTTP headers to be applied to all HTTP requests. |
+| Cookies | Cookies to be sent with each HTTP request. |
+| ClientCertificates | TLS client certificates to send when connecting over HTTP (not supported in Xamarin) |
+
+The following code samples demonstrate the C# setting connection options with the `HubConnectionBuilder`:
```

Comment:
The following code sets connection options with ~the~ `HubConnectionBuilder`:

Reply:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.signalr.client.hubconnectionbuilder?view=aspnetcore-2.1